conf/config.neon

```diff
@@ -2157,6 +2157,16 @@ services:
 	-
 		class: PHPStan\Reflection\BetterReflection\SourceStubber\ReflectionSourceStubberFactory
 
+	-
+		class: PHPStan\Reflection\Deprecation\DeprecationResolver
+		arguments:
+			propertyDeprecationProviders: tagged(phpstan.propertyDeprecationProvider)
```

Comment:
`tagged()` is a cool shortcut but I worry it might lead to reference cycles. It's better to have a class that provides access to user extensions and does `$this->container->getServicesByTag()` itself. That breaks those reference cycles.

For example, in the current PR, you can't have an extension that asks for NodeScopeResolver in its constructor.